### ISSUE TITLE
Cannot remove many 2 many object, error thrown

### ISSUE THREAD

**Issue by Looney77** [sequelize/sequelize#2832]
Hi,

I'm currently facing to a problem while removing a many to many self referenced object. See code below, it throws following error : Possibly unhandled TypeError: Object [object SequelizeInstance] has no method 'map'

Am I doing something wrong ? 

```
var Sequelize = require('sequelize')
  , config    = require('./config')
  , sequelize = new Sequelize(config.database, config.username, config.password, config.db);
  
var FS = sequelize.define('FS', {
  name: Sequelize.STRING
});

FS.hasMany(FS, { as: 'Backup', foreignKey: 'ParentFSId', through: 'FSsBackups' });
FS.hasMany(FS, { as: 'Parent', foreignKey: 'FSId', through: 'FSsBackups' });

var chainer = new Sequelize.Utils.QueryChainer,
    fs1 = FS.build({name: 'FS1'}),
    fs2 = FS.build({name: 'FS2'}),
    fs3 = FS.build({name: 'FS3'}),
    fs4 = FS.build({name: 'FS4'}),
    fs5 = FS.build({name: 'FS5'});
  
sequelize.sync({force:true}).on('success', function() {
  console.log('Sync done!');

  chainer.add(fs1.save())
          .add(fs2.save())
          .add(fs3.save())
          .add(fs4.save())
          .add(fs5.save());

  chainer.run().on('success', function() {
    console.log('Chainer successfull');
    fs1.setBackup([fs2, fs3, fs4]).on('success', function() {
      fs2.setBackup([fs1, fs4, fs5]).on('success', function() {
        console.log('Backups set : ');
        FS.find({where: {name: 'FS1'}, include: [{model: FS, as:'Backup'}]}).success(function(item) {
          console.log('item: ' + JSON.stringify(item));
          fs3.addBackup(fs1).on('success', function() {
            FS.find({where: {name: 'FS3'}, include: [{model: FS, as:'Backup'}]}).success(function(item) {
              console.log('item: ' + JSON.stringify(item));
              fs1.removeBackup(fs2).on('success', function(item) {
                console.log('Removed fs2 from fs1');
                console.log('item ' + JSON.stringify(item));
              }) 
            })
          })
        })
      })
    })
  })
});
```

Br,

Looney

**Comment by janmeier** [comment]
It seems you are passing an instance to a method that expects an array. 

Try looking at the stack trace to see which method is failing.

**Comment by Looney77** [comment]
Hi thanks for replying ...

Indeed ... just had to pass an array to removeBackup ... so dummy lol.

Thanks a lot,

Looney

**Comment by mickhansen** [comment]
@janmeier @Looney77 Hmm okay, i thought `removeHasManyInstance` supported single instances. In any case it would be a case for either supporting a single instance or a better error message.

**Comment by janmeier** [comment]
@mickhansen I think the issue is that the plural remove function gets the same name as the singular an overwrites it.

**Comment by mickhansen** [comment]
@janmeier ah yeah, that makes sense.

**Comment by janmeier** [comment]
@mickhansen but that code should probably be rewritten, see the issue referenced above

**Comment by BridgeAR** [comment]
This should be fixed in 2.1
